src/query/src/query_engine.rs

```diff
@@ -36,12 +36,12 @@ pub use crate::query_engine::state::QueryEngineState;
 pub trait QueryEngine: Send + Sync {
     fn name(&self) -> &str;
 
-    fn sql_to_statement(&self, sql: &str) -> Result<Statement>;
+    fn sql_to_statement(&self, sql: &str) -> Result<Vec<Statement>>;
```

Comment:
I see lots of `.remove(0)` in the codes, what about providing two APIs, one for executing a single statement, which returns an error if there are multiple statements, and one for multiple statements?

Then you could avoid writing lots of `xxx.remove(0)` and `xxx[0]`.

Reply:
I'm afraid it's difficult for callers to judge which one, the single or multi, to call at this tier. 

But as I mentioned in residual discussions, an alternative way is to keep 1str-1statement mapping at query engine level, and let frontend to ensure multi-statement sql is parsed and split. WDYT?

Reply:
What do you think about only changing the frontend, making it receive multiple SQL at once and execute it to datanode one by one? This might keep the changes small but with the same functionality

Reply:
Yes I'm going to move to that direction because frontend can do this well and keep its backend components free of this issue.